pipelines/nextflow/subworkflows/genome_prepare/main.nf

```diff
@@ -35,71 +34,64 @@ include { CHECK_INTEGRITY } from '../../modules/manifest/integrity.nf'
 include { MANIFEST_STATS } from '../../modules/manifest/manifest_stats.nf'
 
 
-// Run main workflow
+// Main workflow
 workflow GENOME_PREPARE {
+    // We expect every input and output stream to have `meta` as the first element in the form of:
+    //   tuple("accession": accession, "production_name": production_name, "prefix": prefix)
 
     take:
-        genomic_dataset // tuple composed of `meta` with accessions like GCA_XXXXXXX.X (as path) and `genome.json`
-        // genomic_dataset // tuple composed of GCA_XXXXXXX.X (as path) and genome.json
+        genomic_dataset // tuple(`meta`, path("genome.json"))
         output_dir // User specified or default
-        cache_dir
         ncbi_check
 
-    // Main data input to this subworkflow is genomic_dataset tuple
     main:
-        // We expect every input and output stream to have `meta` as the first val in the form of:
-        //   tuple("accession": accession, "production_name": production_name, "prefix": prefix)
-
         // Verify genome.json schema
-        checked_genome = CHECK_JSON_SCHEMA_GENOME(genomic_dataset)
+        checked_genome = CHECK_JSON_SCHEMA_GENOME(genomic_dataset).verified_json
 
-        // Download genome data files. Files may or may not include gene models (gff3) and/or peptides.
-        (download_min, download_opt) = DOWNLOAD_ASM_DATA(checked_genome, cache_dir)
+        // Download genome data files. Files may or may not include gene models (GFF3) and/or peptides.
+        DOWNLOAD_ASM_DATA(checked_genome)
+        download_min = DOWNLOAD_ASM_DATA.out.min_set
+        download_opt = DOWNLOAD_ASM_DATA.out.opt_set
 
-        // Check for presence of annotation related output files, then process gene models data if present
-        if (download_opt) {
```

Comment:
why are we removing `if (dwonload_opt)`?

Reply:
Because the logic of those assemblies with annotation is handled by the channel creation itself (`genome_gff_files = checked_genome.join(unpacked_gff, failOnDuplicate: true)`).

Reply:
ok. cool. thanks!